### PR TITLE
LPC55S69 NS target refactor

### PR DESCRIPTION
**Pull request by agansari**
Refactoring LPC55S69 non-secure target by moving partitioning so _ns version can use it.

### CODE REVIEW

**Comment by galak** [comment]
LGTM, but would like @d3zd3z to take a look at the flash partitions bits.

File: boards/arm/lpcxpresso55s69/doc/index.rst [hunk 1]
```diff
@@ -78,10 +78,11 @@ Other hardware features are not currently enabled such as dual core or secure/no
 Targets available for this board are:
 
 - *lpcxpresso55s69_cpu0* secure (S) address space for CPU0
-- *lpcxpresso55s69_ns* non-secure (NS) address space for CPU0
+- *lpcxpresso55s69_nonsecure* non-secure (NS) address space for CPU0
```
Comment (MaureenHelm) [inline boards/arm/lpcxpresso55s69/doc/index.rst:81]: Is _nonsecure really the convention @ioannisg ? I don't think I've seen this anywhere else.

Comment (microbuilder) [reply boards/arm/lpcxpresso55s69/doc/index.rst:81]: @MaureenHelm That's what we used with the an521 (`mps2_an521_nonsecure`), so I proposed keeping that to @agansari, but if you'd prefer appending something shorter we can change the AN521 build target as well.

Comment (MaureenHelm) [reply boards/arm/lpcxpresso55s69/doc/index.rst:81]: I like the idea of getting them aligned. I think all the nordic boards use ns.

Comment (ioannisg) [reply boards/arm/lpcxpresso55s69/doc/index.rst:81]: Yeap, that's right, the ns (without underscore) what Nordic and MuSca boards use.

**Review by ioannisg — CHANGES_REQUESTED** [review]
~~If it is not causing any issues, I'd rather see the non-secure suffix in Kconfig, DTS, .yaml and board target name to be 'ns'. This aligns with other targets in Zephyr with dual processing environment.~~

sorry, i am wrong, MuSca board is using the _nonsecure suffix. the 'ns' is used in nrf5340, nrf9160 (and friends).

**Review by ioannisg — CHANGES_REQUESTED** [review]
In my opinion, we should not filter out tests with the _nonsecure version of this LPC board; then we have to do it for all board target of this kind: Nordic, MusCa, STM, and SiLabs (shortly).

We should apply a generic principle, of filtering out all targets that are not standalone Zephyr images.
In the case of Cortex-M33 with TrustZone, this applies to all targets that have `CONFIG_TRUSTED_EXECUTION_NONSECURE=y`, because _these_ targets require a Secure firmware counterpart, to be able to boot. I don't think Zephyr testing should be executed on these targets.

Alternatively, we could add some property in the _nonsecure board .yml files, to instruct CI not to include these boards in any tests (for example, do build-only)

Note: builds with CONFIG_TRUSTED_EXECUTION_SECURE=y (such as the QEMU mps2_an521) are standalone-capable and may be included in testing.

**Comment by microbuilder** [comment]
> sorry, i am wrong, MuSca board is using the _nonsecure suffix. the 'ns' is used in nrf5340, nrf9160 (and friends).

So Musca and AN521 boards use `_nonsecure`, Nordic uses `_ns`, both are merged, so we should clearly pick one or the other and generate a PR to harmonize naming. Any votes/opinions, then? I don't have any strong feelings myself, but we should use the right one in this PR at least.

**Comment by MaureenHelm** [comment]
> So Musca and AN521 boards use `_nonsecure`, Nordic uses `_ns`, both are merged, so we should clearly pick one or the other and generate a PR to harmonize naming. Any votes/opinions, then? I don't have any strong feelings myself, but we should use the right one in this PR at least.

I lean toward `_ns` because `_nonsecure` is a little long, but I don't have a strong preference.

**Comment by microbuilder** [comment]
> > So Musca and AN521 boards use `_nonsecure`, Nordic uses `_ns`, both are merged, so we should clearly pick one or the other and generate a PR to harmonize naming. Any votes/opinions, then? I don't have any strong feelings myself, but we should use the right one in this PR at least.
> 
> I lean toward `_ns` because `_nonsecure` is a little long, but I don't have a strong preference.

I agree shorter is likely better, and can put a PR together renaming the AN521 board and documentation, and @agansari can revert to the original `_ns` here (sorry Andrei).

**Comment by agansari** [comment]
@microbuilder @MaureenHelm @ioannisg i'll wait for a decision regarding naming conventions, dropping the renaming commit is easy.

**Comment by microbuilder** [comment]
@agansari Everyone seems to prefer the shorter `_ns` so lets stick with that and I'll update the musca and AN521 boards accordingly.

**Comment by agansari** [comment]
@microbuilder done! I've dropped that commit.

**Comment by microbuilder** [comment]
@agansari You might want to amend the commit message from `boars` to `boards` 🐗

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69.dtsi [hunk 2]
```diff
@@ -90,6 +90,45 @@
...
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x00010000>;
+			read-only;
+		};
+		/*
+		 * The flash starting at 0x10010000 and ending at
```
Comment (ioannisg) [inline boards/arm/lpcxpresso55s69/lpcxpresso55s69.dtsi:109]: @agansari is this reserved for TF-M, or for some other image, or library?

Comment (agansari) [reply boards/arm/lpcxpresso55s69/lpcxpresso55s69.dtsi:109]: @ioannisg not sure who uses it, tfm or mcuboot, when creating this partitioning scheme i followed the template of other devices.
Most devices (aside from Nordic) have a range reserved for applications and are not part of a partition, I see Nordic use the storage partition for applications.

Comment (d3zd3z) [reply boards/arm/lpcxpresso55s69/lpcxpresso55s69.dtsi:109]: I suspect it isn't used for anything general on Zephyr. The intent was to be an area for something like a filesystem, or the app to store some kind of state.

**Review by d3zd3z — APPROVED** [review]
This is a good change. Since the two CPUs share memory, it doesn't really make any sense for the partitions to be different.

**Review by MaureenHelm — APPROVED** [review]
Please rebase and resolve the merge conflict

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69.dtsi (hunk 2 above)
Comment (agansari) [reply boards/arm/lpcxpresso55s69/lpcxpresso55s69.dtsi:109]: Will merge this un-partitioned space into storage partition.

**Comment by agansari** [comment]
@ioannisg can you have another look at this pull?

**Comment by agansari** [comment]
not required anymore, will be replaced by #27780

**Pull request closed without merging**